hardware-testing/hardware_testing/abr_tools/abr_read_logs.py

```diff
@@ -0,0 +1,221 @@
+"""Read ABR run logs and save data to ABR testing csv."""
+from .abr_run_logs import get_run_ids_from_storage, get_unseen_run_ids
+from .error_levels import ERROR_LEVELS_PATH
+from typing import Set, Dict, Tuple, Any, List
+import argparse
+import os
+import csv
+import json
+from datetime import datetime
+
+
+def get_modules(file_results: Dict[str, str]) -> Dict[str, Any]:
+    """Get module IPs and models from run log."""
+    modList = (
+        "heaterShakerModuleV1",
+        "temperatureModuleV2",
+        "magneticBlockV1",
+        "thermocyclerModuleV2",
+    )
+    all_modules = {key: None for key in modList}
+    for module in file_results.get("modules", []):
+        if isinstance(module, dict) and module.get("model") in modList:
+            all_modules[module["model"]] = module.get("serialNumber", "")
+    return all_modules
+
+
+def get_error_info(file_results: Dict[str, Any]) -> Tuple[int, str, str, str, str]:
+    """Determines if errors exist in run log and documents them."""
+    error_levels = []
+    # Read error levels file
+    with open(ERROR_LEVELS_PATH, "r") as error_file:
+        error_levels = list(csv.reader(error_file))
+    num_of_errors = len(file_results["errors"])
+    if num_of_errors == 0:
+        error_type = ""
+        error_code = ""
+        error_instrument = ""
+        error_level = ""
+        return 0, error_type, error_code, error_instrument, error_level
+    if num_of_errors > 0:
+        commands_of_run: List[Dict[str, Any]] = file_results.get("commands", [])
+        run_command_error = commands_of_run[-1]  # type: Dict[str, Any]
+        error_str = len(run_command_error.get("error", ""))  # type: int
```

Comment:
```suggestion
        error_str = len(run_command_error.get("error", ""))
```
(should be inferred)